src/routes.js

```diff
@@ -678,66 +680,55 @@ module.exports = function (app) {
      *               type: string
      *           required:
      *             - templateBase64
-     *             - messageBase64
-     *       - name: api-version
-     *         in: query
-     *         description: API version to use. The current version is 1.0. Previous versions, including passing no version, are deprecated.
-     *         required: false
-     *         type: string
-     *       - name: code
-     *         in: query
-     *         description: 'API key'
-     *         required: false
-     *         type: string
-     *       - name: X-MS-CONVERSION-API-KEY
-     *         in: header
-     *         description: 'API key'
-     *         required: false
-     *         type: string
-     *     responses:
-     *       200:
-     *         description: conversion result and unused segments report
-     *       400:
-     *         description: Bad request
-     *       401:
-     *         description: Unauthorized
-     */
-    app.post('/api/convert/hl7', function (req, res) {
+     *             - srcDataBase64
+    *       - name: api-version
+    *         in: query
+    *         description: API version to use.
+    *         required: false
+    *         type: string
+    *       - name: code
+    *         in: query
+    *         description: 'API key'
+    *         required: false
+    *         type: string
+    *       - name: X-MS-CONVERSION-API-KEY
+    *         in: header
+    *         description: 'API key'
+    *         required: false
+    *         type: string
+    *     responses:
+    *       200:
+    *         description: Converted message
+    *       400:
+    *         description: Bad request
+    *       401:
+    *         description: Unauthorized
+    */
+    app.post('/api/convert/:srcDataType', function (req, res) {
         workerPool.exec({
-            'type': '/api/convert/hl7',
-            'messageBase64': req.body.messageBase64,
+            'type': '/api/convert/:srcDataType',
+            'srcDataType': req.params.srcDataType,
+            'srcDataBase64': req.body.srcDataBase64,
             'templateBase64': req.body.templateBase64,
-            'replacementDictionaryBase64': req.body.replacementDictionaryBase64,
-            'templatesMapBase64': req.body.templatesMapBase64,
+            'templatesOverrideBase64': req.body.templatesOverrideBase64
         }).then((result) => {
             res.status(result.status);
-
-            if (req.query['api-version'] === '1.0') {
-                res.json(result.resultMsg);
-            }
-            else {
-                if (result.status === 200) {
-                    res.json(result.resultMsg.fhirResource);
-                }
-                else {
-                    res.json(result.resultMsg);
-                }
-            }
-
+            res.json(result.resultMsg);
             return;
         });
     });
 
-
     /**
     * @swagger
-    * /api/convert/hl7/{template}:
+    * /api/convert/{template}:
```

Comment:
The endpoint needs to have the srcDataType like the one in the post call below.

Reply:
here template (as returned by api/templates) includes the srcDataType.